controllers/component_image_controller.go

```diff
@@ -107,11 +107,14 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, fmt.Errorf("error reading component: %w", err)
 	}
 
+	quayClient := r.BuildQuayClient()
+	org := r.ReadQuayOrganization()
```

Comment:
This reads the data on each reconcile even that doesn't interact with quay.
Since we need to have new token for the deletion, I suggest to read it in two places: at the beginning of the deletion logic and below in regular reconcilation logic where we are sure that requests to quay will be done.

Reply:
ok, I'm going to instantiate the client just before use